services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DccRevocationListStructureProvider.java

```diff
@@ -0,0 +1,193 @@
+package app.coronawarn.server.services.distribution.assembly.component;
+
+import static app.coronawarn.server.services.distribution.assembly.component.DigitalCertificatesStructureProvider.EXPORT_BIN;
+
+import app.coronawarn.server.common.persistence.domain.RevocationEntry;
+import app.coronawarn.server.common.persistence.service.DccRevocationListService;
+import app.coronawarn.server.services.distribution.assembly.structure.Writable;
+import app.coronawarn.server.services.distribution.assembly.structure.WritableOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.ArchiveOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.decorator.signing.DistributionArchiveSigningDecorator;
+import app.coronawarn.server.services.distribution.assembly.structure.directory.Directory;
+import app.coronawarn.server.services.distribution.assembly.structure.directory.DirectoryOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.file.FileOnDisk;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import app.coronawarn.server.services.distribution.dcc.DccRevocationClient;
+import app.coronawarn.server.services.distribution.dcc.DccRevocationListToProtobufMapping;
+import app.coronawarn.server.services.distribution.dcc.FetchDccListException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.bouncycastle.util.encoders.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DccRevocationListStructureProvider {
+
+  private static final Logger logger = LoggerFactory.getLogger(DccRevocationListStructureProvider.class);
+  private static final String KID_ARCHIVE = "kid";
+  public static final String CHUNK = "chunk";
```

Comment:
how do you decide which string is public and which is private?

Reply:
IDE did it for me. I just extracted constants from a string. Will adjust them now